app/src/main/java/org/fossasia/openevent/api/APIClient.java

```diff
@@ -37,6 +48,15 @@ public APIClient() {
                 .setConverter(new GsonConverter(gson))
                 .setEndpoint(Urls.BASE_GET_URL_ALT)
                 .setLogLevel(RestAdapter.LogLevel.FULL)
+                .setErrorHandler(new ErrorHandler() {
+                    @Override
+                    public Throwable handleError(RetrofitError cause) {
+                        ErrorDialogEvent errEvent = new ErrorDialogEvent(null, null);
+                        errEvent.ErrorHandlerFunction(cause, errEvent);
+                        OpenEventApp.postEventOnUIThread(new ErrorDialogEvent(errEvent.ErrorType, errEvent.ErrorDesc));
+                        return new Exception(cause);
```

Comment:
I don't think we should do that because the error will be posted for all API calls, and this API Client is shared everywhere. The if-else or switch will create only 1 object so I don't really understand what's your point here:

> The reason I created the object errEvent was because all the error dialogs differed only in the type of error and the description